Create ui-c8y-1021-80-0-grid-displaying-dtm-properties-for-custom-column-selection-graft-release-cd.md (#3447)
SoftwareAG/c8y-docs: content/change-logs/device-management/ui-c8y-1021-80-0-grid-displaying-dtm-properties-for-custom-column-selection-graft-release-cd.md

---
date: ""

> title: Grid displaying device type mapping properties for custom column selection

Reviewer: ```suggestion
title: Asset-property picker for custom grid columns
```

product_area: Device management & connectivity
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component--KIsStyzM
    label: Device Management app
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: DM-4563
version: 1021.80.0
---

> In the device management application, users can now select device type mapping properties as custom columns in the grid displaying the devices. Previously, this was not possible and users could only select from a predefined set of properties. With this change, users have more flexibility to tailor the grid to their specific needs by selecting any property defined in the device type mapping. This impacts all users of the device management application who want to customize the grid columns to display device type mapping properties.

Reviewer: ```suggestion
In the data grid views, where custom columns are used the “Add custom column” dialog shows a searchable list of asset-property definitions when the Digital Twin Manager (DTM) asset API is available. Choosing a property automatically sets the column’s header and path, eliminating manual typing.
If the DTM service cannot be reached, the application transparently falls back to the previous dialog, so existing workflows continue to work.
```

---
date: ""

> title: Asset-property picker for custom grid columns

Reviewer: ```suggestion
title: List of asset property definitions for data grid custom columns
```

> title: List of asset property definitions for data grid custom columns

Reviewer 2: ```suggestion
title: List of asset properties available when creating custom columns
```

product_area: Device management & connectivity
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component--KIsStyzM
    label: Device Management app
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: DM-4563
version: 1021.80.0
---

> In the data grid views, where custom columns are used the “Add custom column” dialog shows a searchable list of asset-property definitions when the Digital Twin Manager (DTM) asset API is available. Choosing a property automatically sets the column’s header and path, eliminating manual typing.

Reviewer 3: This sentence is too complicated, language wise. Can you break this up?

Co-Pilot suggested this:

In data grid views, when adding a custom column, you can now search and select asset properties from the Digital Twin Manager (DTM) asset API. The column header and path are filled in automatically. If the DTM service isn’t available, the app uses the previous dialog so you can keep working as before.

Reviewer: That is not correct because select asset properties from the Digital Twin Manager (DTM) asset API does not actually say what we want it to.

I can change it to:

```suggestion
In data grid views, when adding a custom column, you can now search through a list of asset-property definitions when the Digital Twin Manager (DTM) asset API is available. The column header and path are filled in automatically. If the DTM service cannot be reached, the application falls back to the previous dialog, so existing workflows continue to work.
```

> In data grid views, when adding a custom column, you can now search through a list of asset-property definitions when the Digital Twin Manager (DTM) asset API is available. The column header and path are filled in automatically. If the DTM service cannot be reached, the application falls back to the previous dialog, so existing workflows continue to work.

Reviewer 2: ```suggestion
In data grid views, when adding a custom column, you can now search through a list of asset property definitions when the Digital Twin Manager (DTM) asset API is available. The column header and path are filled in automatically. If the DTM service cannot be reached, the application falls back to the previous dialog, so existing workflows continue to work.
```

> If the DTM service cannot be reached, the application transparently falls back to the previous dialog, so existing workflows continue to work.

Reviewer 2: ```suggestion
```
